tests/rst_style.py

```diff
@@ -0,0 +1,153 @@
+    last_delimiter_char_index = -1  # The index of the heading delimiter char in the ordered list.
+    for lno, line in enumerate(lines):
+        if HEADING_DELIMITER_RE.search(line):  # The line is a heading delimiter.
+            delimiter_char = line[0]
+            delimiter_char_index = ALLOWED_HEADING_CHARS.index(delimiter_char)
+            if delimiter_char_index > last_delimiter_char_index + 1:
+                # There is a leap of more than 1 in the chars used for the heading delimiters.
+                last_delimiter_char = ALLOWED_HEADING_CHARS[last_delimiter_char_index] \
+                    if last_delimiter_char_index != -1 else None
+                yield lno + 1, f"the heading delimiter {delimiter_char} is not allowed after a" \
+                               f" heading with {last_delimiter_char} as delimiter; follow this" \
+                               f" order: {', '.join(ALLOWED_HEADING_CHARS)}"
+            last_delimiter_char_index = delimiter_char_index
+
+@sphinxlint.checker('.rst')
+def check_max_one_main_heading(file, lines, options=None):
+    """ Check that there is at most one main heading (h1) per document. """
+    code = "".join(lines)
+    nb_main_headings = len(MAIN_HEADING_RE.findall(code))
+    if nb_main_headings > 1:
+        yield 1, "the document should have only one main heading"
+
+@sphinxlint.checker('.rst')
+def check_heading_delimiters_length(file, lines, options=None):
+    """ Check that heading delimiters have the same length as their heading. """
+    for lno, line in enumerate(lines):
+        if HEADING_DELIMITER_RE.search(line):  # The line is a heading delimiter.
+            if MAIN_HEADING_RE.search(''.join(lines[lno:lno+3])):  # Upper delimiter of h1.
+                heading_lno = lno + 1
+            else:  # Lower delimiter of a heading of any level.
+                heading_lno = lno - 1
+            if len(line.rstrip()) != len(lines[heading_lno].rstrip()):
+                yield lno + 1, "the heading delimiter should have the same length as its heading"
+
+@sphinxlint.checker('.rst')
+def check_heading_spacing(file, lines, options=None):
+    """ Check that headings are preceded and followed by at least one blank line. """
+    for lno, line in enumerate(lines):
+        if HEADING_DELIMITER_RE.search(line):  # The line is a heading delimiter.
+            if MAIN_HEADING_RE.search(''.join(lines[lno:lno+3])):  # Upper delimiter of h1.
+                heading_lno = lno + 1
+                main_heading = True
+            else:
+                heading_lno = lno - 1
+                if MAIN_HEADING_RE.search(''.join(lines[lno-2:lno+1])):  # Lower delimiter of h1.
+                    main_heading = True
+                else:  # Lower delimiter of a heading of level 2-6.
+                    main_heading = False
+            if lines[heading_lno - (2 if main_heading else 1)] != '\n':
+                yield heading_lno + 1, "the heading should be preceded by a blank line"
```

Comment:
```suggestion
                yield heading_lno - 1, "the heading should be preceded by a blank line"
```

Reply:
Actually, it should be `heading_lno+0` because the heading has a problem, not the delimiter.